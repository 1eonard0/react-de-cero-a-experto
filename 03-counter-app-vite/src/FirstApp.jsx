import PropTypes from 'prop-types'
const myName = () => 'Leonardo Toledo';

export const FirstApp = ({username, edad}) => {
    //console.log(props);
  return (
    <>
        <h1 data-testid="username-test-id">{ username }</h1>
        <p>{ edad }</p>
    </>
  )
}

FirstApp.propTypes = {
    username: PropTypes.string.isRequired,
    edad: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    username: "No available username",
    edad: "No available age"
}