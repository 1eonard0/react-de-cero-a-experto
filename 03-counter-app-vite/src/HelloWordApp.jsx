export const HelloWordApp = () => {
  return (
    <h1>Hello World!!</h1>
  )
}
