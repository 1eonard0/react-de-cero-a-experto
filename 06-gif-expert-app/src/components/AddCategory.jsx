import PropTypes from "prop-types";
import { useState } from "react";

export const AddCategory = ({ onAddCategory }) => {
    const [inputValue, setInputValue] = useState('');

    const onInputValueChange = ({ target }) => {
        setInputValue(target.value);
    };

    const onSubmit = (event) =>{
        event.preventDefault();
        console.log(inputValue);
        if(inputValue.trim().length == 0) return;
        //onAddCategory( (categories) => [inputValue, ...categories]);
        onAddCategory(inputValue);
        setInputValue('');
    };

  return (
    <form onSubmit={onSubmit} aria-label="form">
        <input 
            type="text"
            placeholder="Buscar Gift"
            value={inputValue}
            onChange={ onInputValueChange }
        />
    </form>
  )
}

AddCategory.propTypes = {
    onAddCategory : PropTypes.func.isRequired
}
