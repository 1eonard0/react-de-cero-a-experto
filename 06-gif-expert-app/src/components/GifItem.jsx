import PropTypes from 'prop-types';

export const GifItem = ({ title, url }) => {
  return (
    <div className="card">
        <img src={ url } alt={ title } />
        <p>{ title }</p>
    </div>
  )
}


/*
Tarea
1. Agregar las propTypes pata title y url 
2. agregar la validaci√≥n para el snapshot
*/

GifItem.propTypes = {
    title : PropTypes.string.isRequired,
    url : PropTypes.string.isRequired
}


