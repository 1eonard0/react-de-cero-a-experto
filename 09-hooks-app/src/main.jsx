import React from 'react';
import ReactDOM from 'react-dom/client';
import './index.css'
//import { HooksApp } from './HooksApp.jsx';
//import { CounterApp } from './01-useState/CounterApp.jsx';
//import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook.jsx';
//import { SimpleForm } from './02-useEffect/SimpleForm.jsx';
//import { MultiplesCustomHooks } from './03-examples/MultiplesCustomHooks.jsx';
//import { Padre } from './07-tarea-memo/Padre.jsx';
import { TodoApp } from './08-use-reducer/TodoApp.jsx';

ReactDOM.createRoot(document.getElementById('root')).render(
  // <React.StrictMode>
    <TodoApp />
  // </React.StrictMode>,
)
