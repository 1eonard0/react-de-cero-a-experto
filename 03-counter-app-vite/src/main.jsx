import React from 'react';
import ReactDOM from 'react-dom/client';
//import { HelloWordApp } from './HelloWordApp';
import { FirstApp } from './FirstApp';
import { CounterApp } from './CounterApp';

import './styles.css';


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp value={2}/>
        {/* <FirstApp username='Leonardo Toledo' edad='32'/> */}
    </React.StrictMode>
);