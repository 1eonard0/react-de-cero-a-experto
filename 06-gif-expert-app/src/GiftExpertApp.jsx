import { useState } from "react"
import { AddCategory, GiftGrid } from "./components";

export const GiftExpertApp = () => {

    const [categories, setCategories] = useState(['Megaman']);

    const addNewCategory = (newCategory) =>{
        setCategories(categories => [newCategory, ...categories]);
    }

  return (
    <>
        <h1>GiftExpertApp</h1>

        <AddCategory onAddCategory={ (value) => addNewCategory(value) } />

        
        { 
          categories.map( category => <GiftGrid key={category} category={category} />) 
        }
        
    </>
  )
}
