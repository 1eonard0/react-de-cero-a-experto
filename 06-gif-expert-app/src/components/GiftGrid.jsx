import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from 'prop-types';

export const GiftGrid = ({ category }) => {

    const { imgs } = useFetchGifs(category); 

  return (
    <div>
        <h1>{ category }</h1>
        <div className="card-grid">
            {
                imgs.map( (img) => 
                <GifItem key={img.id} {...img}/>
                )
            }
        </div>
    </div>
  )
}

GiftGrid.propTypes = {
  category : PropTypes.string.isRequired
}
