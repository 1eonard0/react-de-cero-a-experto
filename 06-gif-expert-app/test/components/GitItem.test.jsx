import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe('Test unitarios para <GifItem />', () => {
    const title = 'sample title';
    const url = 'http://url.com/';
    test('Debe conicidir con el snapshot', () => {
        const { container } = render(<GifItem title={ title } url={ url }/>);
        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar la imagen con el url y el alt indicado', () => {
        render(<GifItem title={ title } url={ url }/>);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Debe mostrar el titlo dentro del elemento p', () => {
        render(<GifItem title={ title } url={ url }/>);
        expect(screen.getByText(title)).toBeTruthy();
    });
});