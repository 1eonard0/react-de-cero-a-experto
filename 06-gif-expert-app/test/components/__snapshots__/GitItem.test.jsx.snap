// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test unitarios para <GifItem /> Debe conicidir con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="sample title"
      src="http://url.com/"
    />
    <p>
      sample title
    </p>
  </div>
</div>
`;
